### PR TITLE
Connection header improvements 

### PR DESCRIPTION
**Pull request by Victor-IX**
Blender Foundation is currently blocking requests to [download.blender.org](https://download.blender.org/). 
I improved the header and included a User ID (UUID4), so they can better monitor the workload caused by the Blender Launcher per user and block users causing too many requests.

### CODE REVIEW

**Comment by zeptofine** [comment]
very cool, I didn't consider this as a possibility. but wouldn't being able to edit the id graphically make it really easy for people to bypass being blocked? or maybe I'm missing the point of the ids

**Comment by Victor-IX** [comment]
Yes, it would be easy for someone to switch the User ID if blocked, but if the goal is to create a way for the Blender Foundation to better monitor the Blender Launcher's web requests and block potentially malicious users, this could still be effective.

For example, they could automatically block user IDs that exceed a specific number of requests per second. Even if the user creates a new ID, it would be blocked quickly without affecting all users. 

Additionally, they could implement a rate limiter per user, ensuring that no user can exceed a certain number of requests per second. This would help prevent abuse while still allowing normal usage.

If we want we can hide this settings from the settings menu

**Comment by zeptofine** [comment]
ah yes, thanks for clarifying. that makes sense. with that context It doesn't need to be hidden

File: source/modules/connection_manager.py [hunk 1]
```diff
@@ -47,8 +51,9 @@ def __init__(self, version: Version, proxy_type=None) -> None:
         self.manager: REQUEST_MANAGER | None = None
 
         # Basic Headers
-        self._headers = {"user-agent": f"Blender-Launcher-v2/{self.version!s} ({get_platform_full()})"}
-
+        # TODO: Change back BLV2 to Blender-Launcher-v2 when it will be unblocked
```
Comment (zeptofine) [inline source/modules/connection_manager.py:54]: Do we plan to apply this before or after the merge?

Comment (Victor-IX) [reply source/modules/connection_manager.py]: Oh yeah, I forgot to swap it back. Good thing I added the comment!

**Comment by mdogancay** [comment]
There is a definite and effective solution to this. I don't think anyone needs to check for updates more than once a day. We can set the update check in the program to be at most once a day. This way, if someone sends frequent requests to Blender Foundation, it is definitely for bad purposes. Also, no one can blame this software...

**Comment by xan2622** [comment]
I agree with @mdogancay. No one needs to check for available builds more than once a day.

I suggest to remove the *Check Automatically* and *On startup* options:
 image
 
This would leave only one possibility for the user: clicking on the [Check] button at the bottom left side of the Blender Launcher UI. Clicking this button would only be possible one per day.

**Comment by Victor-IX** [comment]
Currently, the automatic build check is disabled in the code, so this option does nothing. The goal of this PR was to provide the Blender Foundation with more information about user requests and to differentiate the request headers by user instead of having the same header for all users. Since this program is open source, if malicious people want to use the launcher to create a high load on the Blender server, they can. The high load caused by the launcher under normal usage has been fixed, so this is no longer an issue (very low number of requests when checking for new builds). We could potentially re-enable the automatic check, as it can be useful for some users, but with a minimum interval of at least 6 hours.

**Pull request merged**